### ISSUE TITLE
Positional-only parameters make errors about unpacking dicts mysteriously disappear

### ISSUE THREAD

**Issue by alexmojaki** [microsoft/pyright#7271]
```python
def with_slash(x: int, /, y: str):
    pass


def without_slash(x: int, y: str):
    pass


with_slash(1, **{'z': None})  # no error - why not?
without_slash(2, **{'z': None})  # error: Argument of type "None" cannot be assigned to parameter "y" of type "str"
without_slash(x=3, **{'z': None})  # error: Argument of type "None" cannot be assigned to parameter "y" of type "str"
```

The line `with_slash(1, **{'z': None})` should give the same error as the other two since there's no way to match `**{'z': None}` with `y: str`. The `/` shouldn't make a difference.

**Comment by alexmojaki** [comment]
```python
def func1(x, /, y: str):
    pass


def func2(x, /, y: str, z: str):
    pass


func1(1, **{'a': 2})  # no error - why not?
func2(1, **{'a': 2})  # error: Argument of type "int" cannot be assigned to parameter "z" of type "str"
```

This seems like it might have the same underlying cause.

**Comment by erictraut** [comment]
I agree this is a false negative. I'll investigate further.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.352, which I just published.
